src/dotnet/Parser.cs

```diff
@@ -11,6 +11,10 @@ namespace Microsoft.DotNet.Cli
 {
     public static class Parser
     {
+        // This is used for command line completion because an empty help text will hide commands/options.
+        // This should not be used for actual command/option descriptions.
+        internal const string NoDescription = "-";
```

Comment:
i guess so.. is that true? this sounds hacky but if it is there already...

Reply:
Without knowing many details about completion and the parser, I am having difficulty understanding what  "This should not be used for actual command/option descriptions" implies for maintainers of the code. Naively, everywhere NoDescription is passed reads like an "actual" description to me. I don't know how to review for misuses of NoDescription.

Reply:
I was able to remove the "fake" parser for the `new` command.  The pro is that we no longer have to keep this in sync with what the templating engine provides.  The con is that we lose the ability to auto-complete all the template names (we hard-coded this prior on our side), as the templating engine doesn't provide completion for the template names.

Thoughts?  I'd advocate for removal.

Reply:
Also, I get the "actual" thing now. Could you put a bit more detail in the comment. Basically, expand on the cases where this is safe to use: when the parser is used for complete/telemetry only and external --help wouldn't use it anyway.

Reply:
My next push will change it to:

```
// This is used for descriptions of commands and options that are only defined for `dotnet complete` (i.e. command line completion).
        // For example, a NuGet assembly handles parsing the `nuget` command and options.
        // To get completion for such a command, we have to define a parser that is used for the completion.
        // Command and option help text cannot be empty, otherwise the parser will hide them from the completion list.
        // The value of `-` has no special meaning; it simply prevents these commands and options from being hidden.
        internal const string CompletionOnlyDescription = "-";
```

Reply:
That's great! Thanks!

Reply:
Regarding the template name completions, we only had a partial list (<50% of the out-of-the-box templates) implemented before.  I'd like to remove the duplicate `new` parser now and file a 2.1.4xx issue on dotnet/templating to provide the auto-completion as a proper feature.

Reply:
That's what I've now pushed up; if you strongly disagree with that plan, I'll put it back in.

Reply:
> remove the duplicate new parser now and file a 2.1.4xx issue on dotnet/templating to provide the auto-completion as a proper 

i think it is fine. --help is not shown at all. and auto complete is not well adapted. A breaking change there should be ok